internal/services/controller/healthz.go

```diff
@@ -0,0 +1,66 @@
+package controller
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"castai-agent/internal/config"
+)
+
+func NewHealthzProvider(cfg config.Config) *HealthzProvider {
+	return &HealthzProvider{
+		cfg:             cfg,
+		initHardTimeout: cfg.Controller.PrepTimeout + cfg.Controller.InitialSleepDuration + cfg.Controller.InitializationTimeoutExtension,
+	}
+}
+
+type HealthzProvider struct {
+	cfg             config.Config
+	initHardTimeout time.Duration
+
+	initializeStarted *time.Time
+	lastHealthyAction *time.Time
```

Comment:
nitpick, was finding this at other part of the code the type was not visible, thought it is actually action not a timestamp, would it make sense to name it `lastHealthyActionAt`?